### ISSUE TITLE
Pixel art and File Recursion

### ISSUE THREAD

**Issue by angry-orange** [Particle1904/DatasetHelpers#16]
Hi, thanks for sharing your tool.
Would it be possible to remove the blur from images to work with pixel art? I've already tried with an image in RGB and another in RGBA and the effect was the same. For example:

Current Behavior

![current status](https://github.com/Particle1904/DatasetHelpers/assets/27979775/14d296d9-cb29-40a6-84b6-60db6033560f)

Desired Behavior

![desired status](https://github.com/Particle1904/DatasetHelpers/assets/27979775/877c2949-a0b5-4203-9763-968a473f1584)

Another request would be the possibility of passing a directory and getting all the files (for example: png, txt or etc.) recursively.

**Comment by Particle1904** [comment]
Regarding your request to remove the blur from images, could you please clarify what you mean by "remove the blur"? I haven't implemented a specific feature for removing blur, so I want to make sure I understand your request correctly. If you're referring to the sharpening effect applied by the Sharpen Sigma operation, it's designed to enhance edges and is typically used when downscaling very large images to lower resolutions. However, it may not be suitable for pixel art or images where a smooth appearance is desired.

As for your second request, .NET does provide functionality to search for files within a directory and its subdirectories recursively. While this feature could be useful for finding files of various types, I haven't implemented it in the tool yet. One reason for this is that it would require managing multiple output folders, which adds complexity to the tool's design.

If you have any further questions or suggestions, please feel free to let me know. Your input helps improve the tool for everyone.

**Comment by angry-orange** [comment]
> Regarding your request to remove the blur from images, could you please clarify what you mean by "remove the blur"? I haven't implemented a specific feature for removing blur, so I want to make sure I understand your request correctly. If you're referring to the sharpening effect applied by the Sharpen Sigma operation, it's designed to enhance edges and is typically used when downscaling very large images to lower resolutions. However, it may not be suitable for pixel art or images where a smooth appearance is desired.
> 
> As for your second request, .NET does provide functionality to search for files within a directory and its subdirectories recursively. While this feature could be useful for finding files of various types, I haven't implemented it in the tool yet. One reason for this is that it would require managing multiple output folders, which adds complexity to the tool's design.
> 
> If you have any further questions or suggestions, please feel free to let me know. Your input helps improve the tool for everyone.

Particle1904, thank you for your response.
By "remove the blur" I mean this behavior: [How to Import Pixel Art in Godot 4.0](https://www.youtube.com/watch?v=rjlRP6MYhCA).

**Comment by Particle1904** [comment]
Oh I see, it's just the way Avalonia is upscaling the image to fit the space. I'll see what I can do about it.

**Comment by Particle1904** [comment]
I've just implemented a feature where interpolation won't be applied to images smaller than 256x256 pixels. However, I still want interpolation to be applied to general images. I'm considering increasing this threshold to 380x380 pixels, but I want to maintain the ability to interpolate low-quality images that aren't pixel art.

To be honest, I'm not entirely sure how valuable this feature is for a tool primarily designed for processing Image Datasets, but I've decided to include it in the next release (v2.3.0).

In case there's a strong need for it in the future, I'm open to the idea of adding a checkbox in the Editor page, allowing users to control whether interpolation is applied to an image or not.

Feedback and suggestions are always appreciated!

![image](https://github.com/Particle1904/DatasetHelpers/assets/6633799/52cb5ce8-7b50-4fcd-81fe-8a141c2ec5aa)
![image](https://github.com/Particle1904/DatasetHelpers/assets/6633799/a903993b-27b9-4310-a6ca-0d757b07cefc)
